core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/result/iceberg/IcebergDocument.scala

```diff
@@ -17,8 +17,7 @@ import scala.jdk.CollectionConverters._
   * IcebergDocument is used to read and write a set of T as an Iceberg table.
   * It provides iterator-based read methods and supports multiple writers to write to the same table.
   *
-  * - On construction, the table will be created if it does not exist.
-  * - If the table exists, it will be overridden.
+  * The table must exist when constructing the document
```

Comment:
Since you are adding this assumption, better to make the constructor of all documents package-private.

Reply:
done